core/iwasm/common/wasm_c_api.c

```diff
@@ -460,9 +532,17 @@ wasm_store_new(wasm_engine_t *engine)
         return NULL;
     }
 
-    if (!wasm_runtime_init_thread_env()) {
-        LOG_ERROR("init thread environment failed");
-        return NULL;
+#ifdef os_thread_local_attribute
+    if (thread_local_stores_num++ == 0)
+#else
+    if (!retrive_thread_local_store_num(&engine->stores_by_tid,
+                                        os_self_thread()))
+#endif
+    {
+        if (!wasm_runtime_init_thread_env()) {
+            LOG_ERROR("init thread environment failed");
+            return NULL;
+        }
```

Comment:
Not a better way to wasm_runtime_init_thread_env and increase thread_local_stores_num here:
- when malloc store fails or init store fails below (L550 and L562), we don't decrease thread_local_stores_num and wasm_runtime_destroy_thread_env
- when INIT_VEC(store->modules) and INIT_VEC(store->instances) fail (see INIT_VEC macro), we don't decrease thread_local_stores_num and wasm_runtime_destroy_thread_env

Suggest removing code here, remove L549, init wasm_runtime_init_thread_env and increase thread_local_stores_num only when initialize store successfully, at L565.